src/InstantiableEntity.php

```diff
@@ -8,8 +8,12 @@
 
 class InstantiableEntity extends BaseEntity {
 
-	function __construct( $name, array $properties = [] ) {
-		$this->name = $name;
+	protected $name;
+
+	function __construct( $name = null, array $properties = [] ) {
+		if ( isset( $name ) ) {
```

Comment:
is the isset even necessary? `$this->name` defaults to `null`, so does the argument `$name`, so checking that `$name` is not-null so it won't over-write the existing `null` value seems moot and redundant, and you could save two lines by omitting the check.